src/core/public/chrome/chrome_service.tsx

```diff
@@ -158,6 +164,31 @@ export class ChromeService {
         }
         this.collapsibleNavHeaderRender = render;
       },
+      addNavToGroup: (navGroup: ChromeNavGroup, navLink: ChromeRegistrationNavLink) => {
```

Comment:
Since we are adding `...NavLink` items, this should at least be named `addNavLinkToGroup`.

However, the more I think of it, I find it uncomfortable that, as a consumer of this:
* I have to call it numerous times with a `navGroup` object that I make.
* Changes I might make to the `navGroup` will not be picked up.

A better way to mitigate this would be to make
```ts
addNavGroup: (navGroup: ChromeNavGroup, navLinks: [ChromeRegistrationNavLink], overwrite?: boolean)
```
... where the consumer adds a group and its links in one call.
1. We validate that the `navGroup.id` is unique when `overwrite` is false.
2. `navLink.id` is unique in the group.

Reply:
Hi @AMoo-Miki , I get your point but imaging that a group consists of applications from various plugins from various repos, there is not a centralized place to `addNavGroup`.

But I do think making addNavLinksToGroup to accept an array of navLinks makes sense. Will adjust that.

Reply:
Sadly it is not a 2-way door: being a public API, we will only be able to change it if we make a major version bump.

> imaging that there is a group consists of applications from various plugins in various repos, there is not a centralized place to addNavGroup

This makes the problem a bit more interesting.

```ts
private readonly navGroupsMap$ = new BehaviorSubject<Record<string, NavGroupItemInMap>>({
  '': { id: '', title: '', description: '', navLinks: [] }
});

addNavGroup(id: string, Omit<ChromeNavGroup, 'id'>, failIfExists?: boolean = false) {
  if (!id?.trim?.()) throw 'NavGroup `id` is expected to be a non-blank string';
  const groupsMap = { ...this.navGroupsMap$.getValue() };
  if (id in groupsMap) {
    if (failIfExists) throw 'Duplicate NavGroup `id` found';
    return false;
  }

  groupsMap[id] = {
    ...navGroup,
    id,
    navLinks,
  };
  this.navGroupsMap$.next(groupsMap);
}

addNavLinks(links: ChromeRegistrationNavLink[] | ChromeRegistrationNavLink, groupId?: string) {
  /// still working on this comment
}
```